functionary/core/api/v1/views/task.py

```diff
@@ -106,6 +106,11 @@ def create(self, request: Request, *args, **kwargs):
             task.save()
         except (ValidationError, DjangoValidationError) as err:
             raise serializers.ValidationError(serializers.as_serializer_error(err))
+        except S3FileUploadError as err:
```

Comment:
A couple of things here:

* If S3 is not reachable at all, an S3ConnectionError is what gets raised, which isn't covered here.  You can add that here, or probably even better would be to go back and make S3FileUploadError and S3ConnectionError both build off of a base exception called S3Error and handle that here.  That way we have a catch-all for these situations where we want to handle any S3 related error the same way and don't have to call them all out individually (even though "all" is only 2 right now).

* The error message returned here should really be the same one that we show in the UI.  It's an API user facing error, not a backend administrator facing one.  So showing the API user something like `Error: HTTPConnectionPool(host='localhost', port=9000): Max retries exceeded with url...` is not really appropriate.

Reply:
Added parent `S3Error` exception class that `S3ConnectionError` and `S3FileUploadError` extend from, so now the view and API only need to catch the generic `S3Error`. Also updated the error message that those exceptions store such that they contain sensible user-facing error messages, while the logger messages include the actual exception error for administrators.